### DISCUSSION TITLE
Can I use vue-query in a newly created vue instance through vue.extend?

### DISCUSSION THREAD

**Discussion by Jungho-Cheon** [TanStack/query#5530] [Q&A] [answered]
Hello.
I wrote a modal component that renders through the vue.extend function in vue 2.7.
I initialized VuePlugin in the same way as global Vue, but it looks like I can't find the context. Is there a good way?

### code

```ts
const MyComponentConstructor = Vue.extend(ModalComponent);
const vueQueryPluginOptions: VueQueryPluginOptions = {
	queryClientConfig: {
		defaultOptions: {
			queries: {
				refetchOnWindowFocus: false,
				retry: false,
			},
		},
	},
};

MyComponentConstructor.use(VueQueryPlugin, vueQueryPluginOptions);

modalComponent = new MyComponentConstructor();

const div = document.createElement('div');
div.id = 'global-modal';
document.body.appendChild(div);
modalComponent.$mount('#global-modal');
```

### Result
<img width="891" alt="스크린샷 2023-06-05 오후 12 27 46" src="https://github.com/TanStack/query/assets/61958795/d6ce8740-e829-4bf5-a59e-7f54c721d339">

**Comment by DamianOsipiuk** [comment, marked as answer]
To be honest i'm not familiar with how `Vue.extend` works under the hood.
You can always pass `queryClient` as an option to `useQuery` and other hooks if `inject` does not work properly.

**Reply by MohamedFiqow** [reply]
I don't know about your Vue architecture, but I am currently working on a manual for working with Vue in AEM.
